File: src/types/Table.ts
// src/types/Table.ts
export interface Column {
  name: string;
  type: string;
  nullable: boolean;
  primaryKey: boolean;
  foreignKey?: {
    table: string;
    column: string;
  };
}

export interface Table {
  name: string;
  columns: Column[];
}

export interface Relationship {
  from: {
    table: string;
    column: string;
  };
  to: {
    table: string;
    column: string;
  };
  type: 'ONE_TO_ONE' | 'ONE_TO_MANY' | 'MANY_TO_ONE' | 'MANY_TO_MANY';
}

export interface DatabaseSchema {
  tables: Table[];
  relationships: Relationship[];
}

File: apps/api/src/schema_design/domain/table.py
from dataclasses import dataclass, field


@dataclass(frozen=True)
class Column:
    name: str
    type: str
    nullable: bool
    primary_key: bool
    unique: bool = False


@dataclass
class Table:
    name: str
    columns: list[Column] = field(default_factory=list)
    unique_constraints: list[tuple[str, ...]] = field(default_factory=list)

    def find_column(self, name: str) -> Column | None:
        for column in self.columns:
            if column.name == name:
                return column
        return None
